spec/unit/semver_spec.rb

```diff
@@ -212,6 +212,17 @@
       expect(version.tiny).to    eq(5)
       expect(version.special).to eq('-beta6')
     end
+
+    it 'should not log a deprecation warning when strict == off' do
+      Puppet[:strict] = :off
+      SemVer.new('1.2.3')
+      expect(@logs).to be_empty
+    end
+
+    it 'should log a deprecation warning unless strict == off' do
```

Comment:
I don't think it needs to, as long as default isn't off (which it isn't). That's also what the title suggests. A warning is printed for both warning and error.